x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -937,7 +941,9 @@ export class TaskRunner<
         return { interval: retryInterval };
       }),
       monitoring: this.ruleMonitoring.getMonitoring(),
-      hasError: isErr(schedule),
+      ...(isErr(schedule)
+        ? { taskRunError: createTaskRunError(schedule.error, TaskErrorSource.FRAMEWORK) }
```

Comment:
will there be a followup PR to allow different error sources within the alerting task runner? right now all errors will be marked framework?

Reply:
Yes there is (https://github.com/elastic/kibana/issues/168631)